packages/aws-cdk-lib/aws-stepfunctions/lib/states/distributed-map/item-reader.ts

```diff
@@ -68,10 +87,22 @@ export interface S3ObjectsItemReaderProps extends ItemReaderProps {
  * Item Reader configuration for iterating over objects in an S3 bucket
  */
 export class S3ObjectsItemReader implements IItemReader {
+  private readonly _bucket?: IBucket;
+
   /**
    * S3 Bucket containing objects to iterate over
    */
-  readonly bucket: IBucket;
+  public get bucket(): IBucket {
+    if (!this._bucket) {
+      throw new Error('`bucket` is undefined');
+    }
+    return this._bucket;
+  }
```

Comment:
can you explain why you added this getter instead of making the bucket property optional?

Reply:
Yes - as mentioned above we cannot make `bucket` property optional since it is part of output, hence, I followed [Optional properties: how to solve](https://github.com/aws/jsii/blob/main/packages/jsii-diff/BREAKING_CHANGES.md#optional-properties-how-to-solve) suggestion to add this getter.